### PR TITLE
[MRG] new version of train and test

### LINKED ISSUE #4: Test results for DenseNet-161 ?
Hello Huaizheng,

thank you for sharing these models!
I have a short question, I'm running the model on Python 3.9.5, Pytorch 1.7.1 and have a warning/error:

'[W NNPACK.cpp:80] Could not initialize NNPACK! Reason: Unsupported hardware.'

I'm not sure if I can ignore this warning and I'm not sure of my results.
Could you please provide your test results for the DenseNet-161?

### PR DESCRIPTION
**Pull request by STALLAAAA**
1. deprecate old training scripts and utils[Done]： The new train script can do all the jobs the old one can do. close #4 #3 
2. replace test with infer[Done]: replace the old test scipt with a better organized inferenc script with hydra.
3. add tensorboard plugin[DONE]:  now tensorboard logs are saved at `log/tensorboard/` while training and validation
4. readme and requirements are also updated
5. add MoCo v2 Resnet50 finetune result

### CODE REVIEW

File: tools/train.py [hunk 1]
```diff
@@ -1,259 +1,412 @@
...
-def main():
-    os.environ["CUDA_DEVICE_ORDER"] = "PCI_BUS_ID"
-
-    global best_acc1
-    
-    run_time = str(datetime.datetime.now())
-    cfgs = load_configs(CONFIG_FILE)
-
-    # create log dir and weight dir
-    mkdir(cfgs['weight_dir'])
```
Comment (HuaizhengZhang) [inline tools/train.py:40]: where could i get the saved log information?

File: tools/train.py [hunk 2]
```diff
@@ -1,259 +1,412 @@
...
+def get_pil_version():
+    return "\n        Pillow ({})".format(PIL.__version__)
 
-            if i % cfgs['print_freq'] == 0:
-                logger.info(progress.display(i))
 
-        # TODO: this should also be done with the ProgressMeter
-        logger.info(' * Acc@1 {top1.avg:.3f} Acc@5 {top5.avg:.3f}'
-            .format(top1=top1, top5=top5))
+def collect_env_info():
```
Comment (HuaizhengZhang) [inline tools/train.py:398]: I prefer these functions should be in the utils. 


 Before change this, try to understand why I move these functions to different files which is different from the torch examples.


We build tools while torch examples show how to use them. That's why they put all together but this is not what we want.

File: tools/train.py (hunk 1 above)
Comment (HuaizhengZhang) [reply tools/train.py:40]: env? hyperparamters?

I understand that loss is stored in tensorbord.

**Review by HuaizhengZhang — CHANGES_REQUESTED** [review]
Kindly please understand why I build utils to manage functions before mimic styles from torch examples which are must be in the same file.

Logging info should be kept as hyper, env, etc also important

File: tools/train.py (hunk 1 above)
Comment (STALLAAAA) [reply tools/train.py:40]: log info is saved at `/logs` as `conf/train.yaml` 'hydra' indicated.

File: tools/train.py (hunk 2 above)
Comment (STALLAAAA) [reply tools/train.py]: OK, I will move them to utils.

**Comment by STALLAAAA** [comment]
@HuaizhengZhang Changes have been made, have a check plz.

**Review by HuaizhengZhang — APPROVED** [review]
